docs/source/spec/xml.rst

```diff
@@ -1,26 +1,598 @@
-``xmlFlattened``, ``xmlName``, and ``xmlNamespace`` traits when serializing
-payloads.
+This document defines how to bind Smithy shapes to XML documents. Smithy
+defines several traits that are used to influence the serialization of
+shapes with XML based protocols.
 
 .. contents:: Table of contents
     :depth: 2
     :local:
     :backlinks: none
 
 
+----------------------
+Serializing XML shapes
+----------------------
+
+This document provides recommendations on how Smithy structures and
+shapes within structures SHOULD be serialized with XML based protocols;
+however, protocols MAY choose to deviate from these recommendations
+if necessary.
+
+
+Structure and union serialization
+=================================
+
+All XML serialization starts with a structure. The shape name of a structure
+is used as the outermost XML element name. Members of a structure or union
+shape are serialized as nested XML elements where the name of the element is
+the same as the name of the member.
+
+For example, given the following:
+
+.. code-block:: smithy
+
+    structure MyStructure {
+        foo: String,
+    }
+
+The XML serialization is:
+
+.. code-block:: xml
+
+    <MyStructure>
+        <foo>example</foo>
+    </MyStructure>
+
+
+Custom XML element names
+------------------------
+
+Structure/union member element names can be changed using the
+:ref:`xmlname-trait`.
+
+
+XML attributes
+--------------
+
+The :ref:`xmlattribute-trait` is used to serialize a structure
+member as an XML attribute.
+
+
+``xmlName`` on structures
+-------------------------
+
+An ``xmlName`` trait applied to a structure changes the element name of the
+serialized structure; however, it does not influence the serialization of
+members that target it. Given the following:
+
+.. tabs::
+
+    .. code-tab:: smithy
+
+        @xmlName("AStruct")
+        structure A {
+            b: B,
+        }
+
+        @xmlName("BStruct")
+        structure B {
+            hello: String,
+        }
+
+    .. code-tab:: json
+
+        {
+            "smithy": "0.5.0",
+            "shapes": {
+                "smithy.example#A": {
+                    "type": "structure",
+                    "members": {
+                        "b": {
+                            "target": "smithy.example#B"
+                        }
+                    },
+                    "traits": {
+                        "smithy.api#xmlName": "AStruct"
+                    }
+                },
+                "smithy.example#B": {
+                    "type": "structure",
+                    "members": {
+                        "hello": {
+                            "target": "smithy.api#String"
+                        }
+                    },
+                    "traits": {
+                        "smithy.api#xmlName": "BStruct"
+                    }
+                }
+            }
+        }
+
+The XML serialization of ``AStruct`` is:
```

Comment:
The name of the structure shape itself is `A`, `AStruct` is the value of the `@xmlName` trait.

Reply:
Because `A` is the outermost structure being serialized, the `xmlName` trait takes effect.

Reply:
I'd read this sentence as referring to the XML serialization of the Smithy structure `smithy.example#A`.